File: test/sliceflowsensfix/straight.py
# straight.py — sliceflowsensfix, the STRAIGHT-LINE control class (Python). Expectations: expect.tsv.


def ps01(a):
    return a


def ps02(a):
    x = a + 1
    sink(x)
    return x


def ps03(a):
    x = 1
    x += a
    x += 2
    return x


def ps04(a):
    x = a
    y = x * 2
    return y + x


def ps05(a):
    x = a
    sink(x
         + 1)
    return x

File: test/sliceflowsensfix/straight.cpp
// straight.cpp — sliceflowsensfix, the STRAIGHT-LINE control class (C++): no branch, no loop, no exit —
// the flow-sensitive answer must equal the source-order one. Expectations: expect.tsv.
void sink( int v );

int cs01( int a )
{
    return a;
}

int cs02( int a )
{
    int x = a + 1;
    sink( x );
    return x;
}

int cs03( int a )
{
    int x = 1;
    x += a;
    x += 2;
    return x;
}

int cs04( int a )
{
    int x;
    x = a;
    int y = x * 2;
    return y + x;
}

int cs05( int a )
{
    int x = a;
    sink( x
          + 1 );
    return x;
}
